engine/src/agents/randomagent.h

```diff
@@ -0,0 +1,68 @@
+/*
+  CrazyAra, a deep learning chess variant engine
+  Copyright (C) 2018       Johannes Czech, Moritz Willig, Alena Beyer
+  Copyright (C) 2019-2020  Johannes Czech
+
+  This program is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+/*
+ * @file: mctsagentrandom.h
+ * Created on 05.2021
+ * @author: BluemlJ
+ *
+ * This agents plays random moves and has nothing todo with MCTS. It inherits from the MCTSAgent to be used in his place.
+ * TODO: Remove the MCTSAgent Inheritance and let him inherit directly from agent
```

Comment:
To be honest, I do not remember why it was a problem. I only know that I had problems using the more general agent. I can try to change that

Reply:
First, it is no problem to let RandomAgent inherit from Agent. 

But it is difficult to let an MCTSAgent play against a non-MCTSAgent in the current state of CrazyAra. The problem can also be seen in the original arena method, where it is currently necessary that both agents inherit the MCTSAgent instead of Agent to work. 

To solve this problem some major changes would be necessary and it was easier to inherit from the MCTSAgent instead